Se planea hacer un modelo predictivo utilizando tecnicas de machine learning para la prediccion de:

La probabilidad de que ocurra un crimen en una zona específicada en la ciudad de Bucaramanga y la probabilidad de la modalidad del crimen.

Clasificar la zona en diferentes categorías de seguridad, como "Alta seguridad", "Moderada seguridad", "Baja seguridad", etc.



LIBRERIAS A USAR

In [2]:

# Pandas para manejo de datos
import pandas as pd


# MatplotLib para grafos
import matplotlib.pyplot as plt

# Seaborn para grafos
import seaborn as sns

# Numpy para manejo de datos
import numpy as np
from numpy import dtype

# MinMaxScaler para normalizacion de datos
from sklearn.preprocessing import MinMaxScaler

CARGAR DATOS

In [3]:
        # Url de los datos 
url = "https://drive.google.com/uc?id=1RoSzovZx4l6CixdVfJTt527_WfCFqPO4&export=download"

# Convertir a DataFrame con pandas
df = pd.read_csv(url)


EDA DE LA TABLA PRINCIPAL


///////////////////// TABLA ////////////////////////////////////

In [4]:

# Mostra el tipo de dato de cada columna
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 135076 entries, 0 to 135075
Data columns (total 19 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   ORDEN                   135076 non-null  int64 
 1   ARMAS_MEDIOS            135076 non-null  object
 2   BARRIOS_HECHO           135076 non-null  object
 3   LATITUD                 135076 non-null  object
 4   LONGITUD                135076 non-null  object
 5   ZONA                    135076 non-null  object
 6   NOM_COMUNA              135076 non-null  object
 7   ANO                     135076 non-null  int64 
 8   MES                     135076 non-null  object
 9   DIA                     135076 non-null  int64 
 10  DIA_SEMANA              135076 non-null  object
 11  DESCRIPCION_CONDUCTA    135076 non-null  object
 12  CONDUCTA                135076 non-null  object
 13  CLASIFICACIONES DELITO  135076 non-null  object
 14  CURSO_DE_VIDA           135076 non-n

In [5]:
# Mostrar las primeras filas del DataFrame
df.head()


ORDEN                 ARMAS_MEDIOS  BARRIOS_HECHO        LATITUD  \
0      1  ARMA BLANCA / CORTOPUNZANTE   BUENOS AIRES  7,170,557,382   
1      2  ARMA BLANCA / CORTOPUNZANTE  CAMPO HERMOSO  7,120,645,358   
2      3  ARMA BLANCA / CORTOPUNZANTE  CAMPO HERMOSO  7,120,645,358   
3      4  ARMA BLANCA / CORTOPUNZANTE      COMUNEROS    715,135,927   
4      5  ARMA BLANCA / CORTOPUNZANTE       GIRARDOT  7,170,557,382   

          LONGITUD    ZONA         NOM_COMUNA   ANO        MES  DIA  \
0      -73,135,108  URBANA      14. MORRORICO  2010  01. ENERO    1   
1       -7,312,605  URBANA  05. GARCÍA ROVIRA  2010  01. ENERO    1   
2       -7,312,605  URBANA  05. GARCÍA ROVIRA  2010  01. ENERO    1   
3  -73,145,704,583  URBANA  03. SAN FRANCISCO  2010  01. ENERO    1   
4      -73,135,108  URBANA     04. OCCIDENTAL  2010  01. ENERO    1   

    DIA_SEMANA               DESCRIPCION_CONDUCTA             CONDUCTA  \
0  05. VIERNES  ARTÍCULO 111. LESIONES PERSONALES  LESIONES PERSONALES   
1  05. VIERNES  ARTÍCULO 111. LESIONES PERSONALES  LESIONES PERSONALES   
2  05. VIERNES  ARTÍCULO 111. LESIONES PERSONALES  LESIONES PERSONALES   
3  05. VIERNES  ARTÍCULO 111. LESIONES PERSONALES  LESIONES PERSONALES   
4  05. VIERNES  ARTÍCULO 111. LESIONES PERSONALES  LESIONES PERSONALES   

  CLASIFICACIONES DELITO CURSO_DE_VIDA ESTADO_CIVIL_PERSONA     GENERO  \
0    LESIONES NO FATALES   05. ADULTEZ          UNION LIBRE  MASCULINO   
1    LESIONES NO FATALES   04. JOVENES              SOLTERO  MASCULINO   
2    LESIONES NO FATALES   04. JOVENES              SOLTERO  MASCULINO   
3    LESIONES NO FATALES   05. ADULTEZ               CASADO  MASCULINO   
4    LESIONES NO FATALES   04. JOVENES          UNION LIBRE  MASCULINO   

  MOVIL_AGRESOR MOVIL_VICTIMA  
0         A PIE         A PIE  
1         A PIE         A PIE  
2         A PIE         A PIE  
3         A PIE         A PIE  
4         A PIE         A PIE

///////////////////// COLUMNAS ////////////////////////////////////


In [6]:
print(f"Categorias y recuento: \n",df.ARMAS_MEDIOS.value_counts())
print("Total de categorias: ",df.ARMAS_MEDIOS.unique().shape)
print("Recuento de valores: ", df.ARMAS_MEDIOS.value_counts().sum())

print("Valores nulos o no reportados:", df.ARMAS_MEDIOS.value_counts()['NO REPORTADO'])

Categorias y recuento: 
 ARMAS_MEDIOS
SIN EMPLEO DE ARMAS                   38265
CONTUNDENTES                          37023
ARMA BLANCA / CORTOPUNZANTE           21890
VEHICULO                              13419
ARMA DE FUEGO                          8947
MOTO                                   7919
LLAVE MAESTRA                          2500
NO REPORTADO                           2113
PALANCAS                               1078
ESCOPOLAMINA                            656
LLAMADA TELEFONICA                      278
PUNZANTES                               248
CORTANTES                               123
PERRO                                    84
REDES SOCIALES                           68
DIRECTA                                  65
ACIDO                                    51
BICICLETA                                45
MIXTA                                    45
LICOR ADULTERADO                         36
CORTOPUNZANTES                           33
GASES                                 

No se encuentran valores "nulos" o "no reportados" en la columna BARRIOS_HECHO y un total de 484 barrios

In [7]:
print(f"Categorias y recuento: \n",df.BARRIOS_HECHO.value_counts())
print("Total de categorias: ",df.BARRIOS_HECHO.unique().shape)
print("Recuento de valores: ", df.BARRIOS_HECHO.value_counts().sum())

Categorias y recuento: 
 BARRIOS_HECHO
CENTRO                11199
SAN FRANCISCO          5944
CABECERA DEL LLANO     5794
LA CONCORDIA           5078
PROVENZA               3447
                      ...  
CAMPO GALAN               1
EL CENTRO                 1
EL PROGRESO               1
VILLABEL                  1
CAMPESTRE                 1
Name: count, Length: 484, dtype: int64
Total de categorias:  (484,)
Recuento de valores:  135076


Se encuentran valores "nulos" en las columnas LATITUD y LONGITUD que aparecen como "xx.xxxx" y "yy.yyyy" respectivamente con un recuento total 6363 y un recuento total de 135076

In [8]:
print("Recuento categorias: ", df.LATITUD.value_counts())
print("Recuento de valores", df.LATITUD.value_counts().sum())
print("/////////////////////////////////////////////////////")
print("Recuento categorias: ", df.LONGITUD.value_counts())
print("Recuento de valores", df.LONGITUD.value_counts().sum())

Recuento categorias:  LATITUD
xx.xxxx          6363
7,178,045,669    1799
7,170,557,382    1375
7,144,371        1163
7,164,623,864    1073
                 ... 
7,104,305,234       1
7,102,639,059       1
7,103,315,112       1
7,127,396,841       1
7,175,989           1
Name: count, Length: 64429, dtype: int64
Recuento de valores 135076
/////////////////////////////////////////////////////
Recuento categorias:  LONGITUD
-yy.yyyy           6363
-73,130,224,418    1799
-73,135,108        1375
-73,128,085        1162
-73,139,140,529    1080
                   ... 
-73,146,044,535       1
-73,143,981,597       1
-73,116,606,589       1
-73,102,834,148       1
-73,130,875           1
Name: count, Length: 60106, dtype: int64
Recuento de valores 135076


Se encuentran valores "nulos" o "no reportados" en la columna ZONA

In [9]:
print(f"Categorias y recuento: \n",df.ZONA.value_counts())
print("Total de categorias: ",df.ZONA.unique().shape)
print("Recuento de valores: ", df.ZONA.value_counts().sum())

print("Valores nulos o no reportados:", df.ARMAS_MEDIOS.value_counts()['NO REPORTADO'])

Categorias y recuento: 
 ZONA
URBANA        131547
NO REPORTA      2189
RURAL           1183
OTROS            157
Name: count, dtype: int64
Total de categorias:  (4,)
Recuento de valores:  135076
Valores nulos o no reportados: 2113


Se encuentran valores "nulos" o "sin informacion" en la columna NOM_COMUNA

In [10]:
print(f"Categorias y recuento: \n",df.NOM_COMUNA.value_counts())
print("Total de categorias: ",df.NOM_COMUNA.unique().shape)
print("Recuento de valores: ", df.NOM_COMUNA.value_counts().sum())

#print("Valores nulos o no reportados:", df.ARMAS_MEDIOS.value_counts()['NO REPORTADO'])

Categorias y recuento: 
 NOM_COMUNA
03. SAN FRANCISCO         15875
15. CENTRO                13872
13. ORIENTAL              13606
12. CABECERA DEL LLANO    12835
01. NORTE                 12708
06. LA CONCORDIA           9994
05. GARCÍA ROVIRA          8959
04. OCCIDENTAL             7736
10. PROVENZA               6639
09. LA PEDREGOSA           4994
02. NOR ORIENTAL           4654
11. SUR                    4532
17. MUTIS                  4196
08. SUR OCCIDENTE          3419
14. MORRORICO              3290
07. LA CIUDADELA           3247
16. LAGOS DEL CACIQUE      3228
CORREGIMIENTO 1             577
CORREGIMIENTO 3             404
FLORIDABLANCA               121
CORREGIMIENTO 2             100
SIN INFORMACION              45
GIRON                        38
MEDELLIN                      3
PIEDECUESTA                   3
BARRANCABERMEJA               1
Name: count, dtype: int64
Total de categorias:  (26,)
Recuento de valores:  135076


Sin discrepancias en las fechas ni valores nulos, datos hasta el 2021

In [11]:
print("Recuento años:", df.ANO.value_counts())
print("Total años:", df.ANO.value_counts().sum())
print("/////////////////////////////////////")
print("Recuento meses:", df.MES.value_counts())
print("Total meses:", df.MES.value_counts().sum())
print("/////////////////////////////////////")
print("Recuento dias:", df.DIA.value_counts())
print("Total dias:", df.DIA.value_counts().sum())
print("/////////////////////////////////////")
print("Recuento dia de la semana:", df.DIA_SEMANA.value_counts())
print("Total dia de la semana:", df.DIA_SEMANA.value_counts().sum())

Recuento años: ANO
2016    13610
2018    13492
2019    12566
2017    12308
2021    11721
2012    10912
2015    10829
2013    10759
2011    10367
2014    10214
2020     9605
2010     8693
Name: count, dtype: int64
Total años: 135076
/////////////////////////////////////
Recuento meses: MES
10. OCTUBRE       11990
12. DICIEMBRE     11774
01. ENERO         11563
02. FEBRERO       11476
09. SEPTIEMBRE    11348
08. AGOSTO        11251
03. MARZO         11177
07. JULIO         11146
05. MAYO          11124
11. NOVIEMBRE     10781
04. ABRIL         10748
06. JUNIO         10698
Name: count, dtype: int64
Total meses: 135076
/////////////////////////////////////
Recuento dias: DIA
1     5097
10    4723
2     4588
5     4575
9     4556
3     4519
12    4495
15    4493
14    4492
4     4489
16    4485
22    4481
8     4455
23    4441
7     4436
17    4423
18    4413
20    4405
13    4399
6     4383
28    4376
11    4375
24    4353
27    4346
25    4308
21    4293
19    4275
26    4179
30    3967


Se encuentran valores "nulos" o "no reportados" en la columna DESCRIPCION_CONDUCTA con un total de 12 datos

In [12]:
print(f"Categorias y recuento: \n",df.DESCRIPCION_CONDUCTA.value_counts())
print("Total de categorias: ",df.DESCRIPCION_CONDUCTA.unique().shape)
print("Recuento de valores: ", df.DESCRIPCION_CONDUCTA.value_counts().sum())

print("Valores nulos o no reportados:", df.DESCRIPCION_CONDUCTA.value_counts()['NO REPORTA'])

Categorias y recuento: 
 DESCRIPCION_CONDUCTA
ARTÍCULO 239. HURTO A PERSONAS                                                                                     48452
ARTÍCULO 111. LESIONES PERSONALES                                                                                  24450
ARTÍCULO 120. LESIONES CULPOSAS ( EN ACCIDENTE DE TRANSITO )                                                       21128
ARTÍCULO 229. VIOLENCIA INTRAFAMILIAR                                                                              16519
ARTÍCULO 239. HURTO A ENTIDADES COMERCIALES                                                                         8937
ARTÍCULO 239. HURTO A RESIDENCIAS                                                                                   4752
ARTÍCULO 239. HURTO A MOTOCICLETAS                                                                                  3708
ARTÍCULO 209. ACTOS SEXUALES CON MENOR DE 14 AÑOS                                                          

Se encuentran valores "nulos" o "no reportados" en la columna CONDUCTA con un total de 12 datos y una discrepancia respecto a DESCRIPCION_CONDUCTA con un total de 41 categorias frente a 43

In [13]:
print(f"Categorias y recuento: \n",df.CONDUCTA.value_counts())
print("Total de categorias: ",df.CONDUCTA.unique().shape)
print("Recuento de valores: ", df.CONDUCTA.value_counts().sum())

print("Valores nulos o no reportados:", df.CONDUCTA.value_counts()['NO REPORTA'])

Categorias y recuento: 
 CONDUCTA
HURTO A PERSONAS                                                                                   48452
LESIONES PERSONALES                                                                                24450
LESIONES CULPOSAS ( EN ACCIDENTE DE TRANSITO )                                                     21130
VIOLENCIA INTRAFAMILIAR                                                                            16519
HURTO A ENTIDADES COMERCIALES                                                                       8937
HURTO A RESIDENCIAS                                                                                 4752
HURTO A MOTOCICLETAS                                                                                3708
ACTOS SEXUALES CON MENOR DE 14 AÑOS                                                                 1529
HOMICIDIO                                                                                           1365
ACCESO CARNAL ABUSIVO

Se encuentran valores "nulos" o "no reportados" en la columna CLASIFICACIONES DELITO con un total de 12 datos

In [14]:
print(f"Categorias y recuento: \n",df['CLASIFICACIONES DELITO'].value_counts())
print("Total de categorias: ",df['CLASIFICACIONES DELITO'].unique().shape)
print("Recuento de valores: ", df['CLASIFICACIONES DELITO'].value_counts().sum())

print("Valores nulos o no reportados:", df['CLASIFICACIONES DELITO'].value_counts()['NO REPORTA'])

Categorias y recuento: 
 CLASIFICACIONES DELITO
LESIONES NO FATALES    128993
VIOLENCIA SEXUAL         4047
LESIONES FATALES         2024
NO REPORTA                 12
Name: count, dtype: int64
Total de categorias:  (4,)
Recuento de valores:  135076
Valores nulos o no reportados: 12


Se encuentran valores "nulos" o "no reportados" en la columna MOVIL_VICTIMA y MOVIL_AGRESOR con un total de 881 datos y 434 datos respectivamente

In [15]:
print(f"Categorias y recuento: \n",df.MOVIL_VICTIMA.value_counts())
print("Total de categorias: ",df.MOVIL_VICTIMA.unique().shape)
print("Recuento de valores: ", df.MOVIL_VICTIMA.value_counts().sum())

print("Valores nulos o no reportados:", df.MOVIL_VICTIMA.value_counts()['NO REPORTA'])

print("/////////////////////////////////////")

print(f"Categorias y recuento: \n",df.MOVIL_AGRESOR.value_counts())
print("Total de categorias: ",df.MOVIL_AGRESOR.unique().shape)
print("Recuento de valores: ", df.MOVIL_AGRESOR.value_counts().sum())

print("Valores nulos o no reportados:", df.MOVIL_AGRESOR.value_counts()['NO REPORTA'])

Categorias y recuento: 
 MOVIL_VICTIMA
A PIE                    104797
CONDUCTOR MOTOCICLETA     16915
CONDUCTOR VEHICULO         5457
PASAJERO BUS               1877
PASAJERO MOTOCICLETA       1250
BICICLETA                  1126
CONDUCTOR TAXI             1108
NO REPORTA                  881
PASAJERO TAXI               795
PASAJERO VEHICULO           471
CONDUCTOR BUS               279
PASAJERO METRO              117
PASAJERO AERONAVE             2
PASAJERO BARCO                1
Name: count, dtype: int64
Total de categorias:  (14,)
Recuento de valores:  135076
Valores nulos o no reportados: 881
/////////////////////////////////////
Categorias y recuento: 
 MOVIL_AGRESOR
A PIE                    93438
CONDUCTOR MOTOCICLETA    15067
CONDUCTOR VEHICULO       13022
PASAJERO MOTOCICLETA      7339
CONDUCTOR TAXI            2931
PASAJERO BUS              1547
PASAJERO TAXI              569
NO REPORTA                 434
BICICLETA                  366
PASAJERO VEHICULO          185
PASAJERO

FEATURE ENGINEERING


///////////////////// ELIMINACION COLUMNAS ////////////////////////////////////

Se eliminan las columnas ESTADO_CIVIL_PERSONA, CURSO_DE_VIDA, GENERO, ZONA por falta de aportar informacion relevante para el modelo
y se elimina la columna CONDUCTA por redundancia con la columna DESCRIPCION_CONDUCTA y una discrepancia en los datos

Tambien se elimina NOM_COMUNA por poder obtenerse la misma informacion de la columna BARRIOS_HECHO

In [16]:
df_normalizada = df

In [17]:


df_normalizada = df_normalizada.drop(['ESTADO_CIVIL_PERSONA', 'CURSO_DE_VIDA', 'GENERO', 'CONDUCTA', 'ZONA', 'NOM_COMUNA', 'ORDEN'], axis=1)

///////////////////// RENOMBRAR COLUMNAS ////////////////////////////////////

In [18]:
df_normalizada = df_normalizada.rename(columns={'CLASIFICACIONES DELITO': 'CLASIFICACIONES_DELITO'})

///////////////////// TRANSFORMACION DATOS Y COLUMNAS //////////////////////////////////

Se tranforman las columnas ANO, MES, DIA, DIA_SEMANA a una columna llamada FECHA_HECHO con el formato YYYY-MM-DD

Primero se transforma el mes a su equivalente numerico mapeando cada uno de los meses a un numero del 1 al 12

Luego se crea la columna FECHA_TEMP concatenando las columnas ANO, MES_NUM y DIA y se convierte a tipo datetime en una nueva columna llamada FECHA_HECHO

Y se eliminan las columnas ANO, MES, DIA y FECHA_TEMP

En base a FECHA_HECHO se sacan las columnas numericas de ANO, MES y DIA para el posterior entrenamiento

In [19]:
mes_mapping = {
    "01. ENERO": 1, "02. FEBRERO": 2, "03. MARZO": 3, "04. ABRIL": 4,
    "05. MAYO": 5, "06. JUNIO": 6, "07. JULIO": 7, "08. AGOSTO": 8,
    "09. SEPTIEMBRE": 9, "10. OCTUBRE": 10, "11. NOVIEMBRE": 11, "12. DICIEMBRE": 12
}
df_normalizada['MES_NUM'] = df['MES'].map(mes_mapping)

In [20]:
df_normalizada['FECHA_TEMP'] = pd.to_datetime(df_normalizada['ANO'].astype(str) + '-' + df_normalizada['MES_NUM'].astype(str) + '-' + df_normalizada['DIA'].astype(str))

df_normalizada['FECHA_HECHO'] = pd.to_datetime(df_normalizada['FECHA_TEMP'], format='%Y-%m-%d', errors='coerce')


In [21]:
df_normalizada = df_normalizada.drop(['ANO', 'MES', 'DIA', 'FECHA_TEMP', 'MES_NUM'], axis=1)

In [22]:
df_normalizada['ANO'] = df_normalizada['FECHA_HECHO'].dt.year
df_normalizada['MES'] = df_normalizada['FECHA_HECHO'].dt.month
df_normalizada['DIA'] = df_normalizada['FECHA_HECHO'].dt.day

In [23]:
df_normalizada = df_normalizada.drop(['FECHA_HECHO'], axis=1)

Se tranforman las columnas LATITUD y LONGITUD a tipo float64 y se imputan los valores "xx.xxxx" y "yy.yyyy" que representan valores nulos con la media 
Y se aplica Min-Max Scaling a las columnas 'LATITUD' y 'LONGITUD' para normalizar los datos

In [24]:
# Se convierten las columnas LATITUD y LONGITUD a tipo float64
df_normalizada['LATITUD'] = pd.to_numeric(df_normalizada['LATITUD'].str.replace(',', ''), errors='coerce')
df_normalizada['LONGITUD'] = pd.to_numeric(df_normalizada['LONGITUD'].str.replace(',', ''), errors='coerce')

# Calcular la media de cada columna (ignorando los NaN)
latitud_mean = df_normalizada['LATITUD'].mean()
longitud_mean = df_normalizada['LONGITUD'].mean()

# Imputar los valores NaN con la media
df_normalizada['LATITUD'] = df_normalizada['LATITUD'].fillna(latitud_mean)
df_normalizada['LONGITUD'] = df_normalizada['LONGITUD'].fillna(longitud_mean)

In [25]:
# Inicializar el escalador
scaler = MinMaxScaler()

# Aplicar Min-Max Scaling a las columnas 'LATITUD' y 'LONGITUD'
df_normalizada[['LATITUD', 'LONGITUD']] = scaler.fit_transform(df_normalizada[['LATITUD', 'LONGITUD']])

///////////////////// NORMALIZACION DE DATOS ////////////////////////////////////

Se aplica one-hot encoding a las columnas ARMAS_MEDIOS, BARRIOS_HECHO, DIA_SEMANA, DESCRIPCION_CONDUCTA, CLASIFICACIONES DELITO, MOVIL_AGRESOR, MOVIL_VICTIMA

In [26]:
df_normalizada = pd.get_dummies(df_normalizada, columns=[
    'ARMAS_MEDIOS',
    'BARRIOS_HECHO',
    'DIA_SEMANA',
    'DESCRIPCION_CONDUCTA',
    'CLASIFICACIONES_DELITO',
    'MOVIL_AGRESOR',
    'MOVIL_VICTIMA'
], drop_first=True)



EDA TABLA NORMALIZADA

In [27]:
df_normalizada.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 135076 entries, 0 to 135075
Columns: 607 entries, LATITUD to MOVIL_VICTIMA_PASAJERO VEHICULO
dtypes: bool(602), float64(2), int32(3)
memory usage: 81.2 MB


In [39]:
df_normalizada.head()

LATITUD  LONGITUD   ANO  MES  DIA  ARMAS_MEDIOS_AGUA CALIENTE  \
0  1.002491e-07   1.00000  2010    1    1                       False   
1  9.955130e-08   1.00000  2010    1    1                       False   
2  9.955130e-08   1.00000  2010    1    1                       False   
3  9.998071e-09   0.99999  2010    1    1                       False   
4  1.002491e-07   1.00000  2010    1    1                       False   

   ARMAS_MEDIOS_ALUCINOGENOS  ARMAS_MEDIOS_ARMA BLANCA / CORTOPUNZANTE  \
0                      False                                      True   
1                      False                                      True   
2                      False                                      True   
3                      False                                      True   
4                      False                                      True   

   ARMAS_MEDIOS_ARMA DE FUEGO  ARMAS_MEDIOS_ARMA TRAUMATICA  ...  \
0                       False                         False  ...   
1                       False                         False  ...   
2                       False                         False  ...   
3                       False                         False  ...   
4                       False                         False  ...   

   MOVIL_VICTIMA_PASAJERO AERONAVE  MOVIL_VICTIMA_PASAJERO BARCO  \
0                            False                         False   
1                            False                         False   
2                            False                         False   
3                            False                         False   
4                            False                         False   

   MOVIL_VICTIMA_PASAJERO BUS  MOVIL_VICTIMA_PASAJERO METRO  \
0                       False                         False   
1                       False                         False   
2                       False                         False   
3                       False                         False   
4                       False                         False   

   MOVIL_VICTIMA_PASAJERO MOTOCICLETA  MOVIL_VICTIMA_PASAJERO TAXI  \
0                               False                        False   
1                               False                        False   
2                               False                        False   
3                               False                        False   
4                               False                        False   

   MOVIL_VICTIMA_PASAJERO VEHICULO         BARRIO  TOTAL_INCIDENTES  \
0                            False   BUENOS AIRES               588   
1                            False  CAMPO HERMOSO              2784   
2                            False  CAMPO HERMOSO              2784   
3                            False      COMUNEROS              2235   
4                            False       GIRARDOT              2608   

   CATEGORIA_SEGURIDAD  
0       Alta seguridad  
1   Moderada seguridad  
2   Moderada seguridad  
3   Moderada seguridad  
4   Moderada seguridad  

[5 rows x 610 columns]

In [36]:
df_normalizada.value_counts('CATEGORIA_SEGURIDAD')

CATEGORIA_SEGURIDAD
Moderada seguridad    35362
Alta seguridad        34155
Baja seguridad        31462
Name: count, dtype: int64

In [35]:
df_normalizada['TOTAL_INCIDENTES'].describe()


count    135076.000000
mean       2685.199754
std        3064.885586
min           1.000000
25%         536.000000
50%        1698.000000
75%        3441.000000
max       11199.000000
Name: TOTAL_INCIDENTES, dtype: float64

FEATURE ENGINEERING PARA LA TABLA NORMALIZADA

In [33]:
barrios_columns = df_normalizada.filter(like='BARRIOS_HECHO_').columns
df_normalizada['BARRIO'] = df_normalizada[barrios_columns].idxmax(axis=1).str.replace('BARRIOS_HECHO_', '')


# Contar el número de incidentes por barrio
incidentes_por_barrio = df_normalizada['BARRIO'].value_counts().reset_index()
incidentes_por_barrio.columns = ['BARRIO', 'TOTAL_INCIDENTES']

# Unir este conteo de incidentes con el DataFrame original
df_normalizada = df_normalizada.merge(incidentes_por_barrio, on='BARRIO', how='left')

# Crear la columna CATEGORIA_SEGURIDAD basada en los umbrales de incidentes
df_normalizada['CATEGORIA_SEGURIDAD'] = pd.cut(df_normalizada['TOTAL_INCIDENTES'],
                                               bins=[536, 1698, 3441, float('inf')],
                                               labels=['Alta seguridad', 'Moderada seguridad', 'Baja seguridad'])

# Verificar los primeros resultados
df_normalizada[['BARRIO', 'TOTAL_INCIDENTES', 'CATEGORIA_SEGURIDAD']].head()






BARRIO  TOTAL_INCIDENTES CATEGORIA_SEGURIDAD
0   BUENOS AIRES               588      Alta seguridad
1  CAMPO HERMOSO              2784  Moderada seguridad
2  CAMPO HERMOSO              2784  Moderada seguridad
3      COMUNEROS              2235  Moderada seguridad
4       GIRARDOT              2608  Moderada seguridad

In [38]:
df_normalizada = df_normalizada.drop([barrios_columns], axis=1) 

KeyError: "[('BARRIOS_HECHO_13 DE JUNIO', 'BARRIOS_HECHO_20 DE JULIO', 'BARRIOS_HECHO_23 DE JUNIO', 'BARRIOS_HECHO_5 DE ENERO', 'BARRIOS_HECHO_ACACIAS', 'BARRIOS_HECHO_AFRICA', 'BARRIOS_HECHO_ALARCON', 'BARRIOS_HECHO_ALBANIA', 'BARRIOS_HECHO_ALFONSO LOPEZ', 'BARRIOS_HECHO_ALTO VIENTO', 'BARRIOS_HECHO_ALTO VIENTO I', 'BARRIOS_HECHO_ALTO VIENTO II', 'BARRIOS_HECHO_ALTOS DE CAÑAVERAL CAMPESTRE', 'BARRIOS_HECHO_ALTOS DE CAÑAVERAL I', 'BARRIOS_HECHO_ALTOS DEL CACIQUE', 'BARRIOS_HECHO_ALTOS DEL CAMPESTRE', 'BARRIOS_HECHO_ALTOS DEL KENNEDY', 'BARRIOS_HECHO_ALTOS DEL PROGRESO', 'BARRIOS_HECHO_ALVAREZ', 'BARRIOS_HECHO_ALVAREZ LAS AMERICAS', 'BARRIOS_HECHO_ANDALUCIA', 'BARRIOS_HECHO_ANILLO VIAL', 'BARRIOS_HECHO_ANTIGUA COLOMBIA', 'BARRIOS_HECHO_ANTIGUO CAMPESTRE', 'BARRIOS_HECHO_ANTIGUO COLOMBIA', 'BARRIOS_HECHO_ANTONIA SANTOS', 'BARRIOS_HECHO_ANTONIA SANTOS CENTRO', 'BARRIOS_HECHO_ANTONIA SANTOS SUR', 'BARRIOS_HECHO_ASENTAMIENTO SERVIUNON', 'BARRIOS_HECHO_ASENTAMIENTO SERVUNION', 'BARRIOS_HECHO_ASTURIAS', 'BARRIOS_HECHO_AUTOPISTA F/BLANCA-P/CUESTA.', 'BARRIOS_HECHO_AUTOPISTA P/CUESTA - F/BLANCA.', 'BARRIOS_HECHO_BALCON DEL LAGO', 'BARRIOS_HECHO_BALCON DEL TEJAR', 'BARRIOS_HECHO_BALCONCITOS', 'BARRIOS_HECHO_BALCONES DEL KENNEDY', 'BARRIOS_HECHO_BALCONES DEL SUR', 'BARRIOS_HECHO_BARRIO NUEVO', 'BARRIOS_HECHO_BAVARIA', 'BARRIOS_HECHO_BELLAVISTA', 'BARRIOS_HECHO_BELLAVISTA CASAS', 'BARRIOS_HECHO_BELLAVISTA FLORIDA', 'BARRIOS_HECHO_BETANIA', 'BARRIOS_HECHO_BOLARQUI', 'BARRIOS_HECHO_BOLIVAR', 'BARRIOS_HECHO_BONANZA CAMPESTRE (COLORADOS)', 'BARRIOS_HECHO_BOSCONIA', 'BARRIOS_HECHO_BOSQUE NORTE', 'BARRIOS_HECHO_BOSQUES DEL PAYADOR', 'BARRIOS_HECHO_BOULEVAR DEL CACIQUE', 'BARRIOS_HECHO_BRISAS DE PROVENZA', 'BARRIOS_HECHO_BRISAS DEL CAMPO', 'BARRIOS_HECHO_BRISAS DEL MUTIS', 'BARRIOS_HECHO_BRISAS DEL PALMAR', 'BARRIOS_HECHO_BRISAS DEL PARAISO', 'BARRIOS_HECHO_BUCARAMANGA', 'BARRIOS_HECHO_BUENAVISTA', 'BARRIOS_HECHO_BUENOS AIRES', 'BARRIOS_HECHO_BULEVAR DEL CACIQUE', 'BARRIOS_HECHO_C.R. SERREZUELA', 'BARRIOS_HECHO_C.R. SOLERI', 'BARRIOS_HECHO_C.R. VERSALLES REAL', 'BARRIOS_HECHO_CABECERA DEL LLANO', 'BARRIOS_HECHO_CACIQUE CONDOMINIO', 'BARRIOS_HECHO_CACIQUE IMPERIAL', 'BARRIOS_HECHO_CACIQUE IPANA', 'BARRIOS_HECHO_CAFE MADRID', 'BARRIOS_HECHO_CALDAS', 'BARRIOS_HECHO_CALLE REAL', 'BARRIOS_HECHO_CAMILO TORRES', 'BARRIOS_HECHO_CAMPESTRE', 'BARRIOS_HECHO_CAMPESTRE NORTE (COLORADOS)', 'BARRIOS_HECHO_CAMPO GALAN', 'BARRIOS_HECHO_CAMPO HERMOSO', 'BARRIOS_HECHO_CAMPO MADRID', 'BARRIOS_HECHO_CAMPO VALDES 1  C-4', 'BARRIOS_HECHO_CAMPO VERDE', 'BARRIOS_HECHO_CANDILES', 'BARRIOS_HECHO_CANELOS', 'BARRIOS_HECHO_CARLOS PIZARRO', 'BARRIOS_HECHO_CARRASCO', 'BARRIOS_HECHO_CASCO ANTIGUO DE FLORIDABLANCA', 'BARRIOS_HECHO_CAÑAVERAL', 'BARRIOS_HECHO_CENFER', 'BARRIOS_HECHO_CENTAUROS', 'BARRIOS_HECHO_CENTRAL DE ABASTOS', 'BARRIOS_HECHO_CENTRO', 'BARRIOS_HECHO_CENTRO P/TA', 'BARRIOS_HECHO_CHAPINERO', 'BARRIOS_HECHO_CHIMITA', 'BARRIOS_HECHO_CHORRERAS DE DON JUAN', 'BARRIOS_HECHO_CIUDAD METROPOLITANA', 'BARRIOS_HECHO_CIUDAD VENECIA', 'BARRIOS_HECHO_CIUDADELA LOS PRINCIPES', 'BARRIOS_HECHO_CIUDADELA REAL DE MINAS', 'BARRIOS_HECHO_CLAVERIANO', 'BARRIOS_HECHO_COLORADOS', 'BARRIOS_HECHO_COLSEGUROS NORTE', 'BARRIOS_HECHO_COMUNEROS', 'BARRIOS_HECHO_CONDADO CAMPESTRE', 'BARRIOS_HECHO_CONDADO DE GIBRALTAR', 'BARRIOS_HECHO_CONQUISTADORES', 'BARRIOS_HECHO_CONUCOS', 'BARRIOS_HECHO_CORDONCILLO I', 'BARRIOS_HECHO_CORDONCILLO II', 'BARRIOS_HECHO_DANGOND', 'BARRIOS_HECHO_DELICIAS', 'BARRIOS_HECHO_DELICIAS ALTAS', 'BARRIOS_HECHO_DIAMANTE I', 'BARRIOS_HECHO_DIAMANTE II', 'BARRIOS_HECHO_DIVINO NIÑO (COLORADOS)', 'BARRIOS_HECHO_DON BOSCO', 'BARRIOS_HECHO_EL BOSQUESITO', 'BARRIOS_HECHO_EL CARMEN I', 'BARRIOS_HECHO_EL CARMEN V', 'BARRIOS_HECHO_EL CENTRO', 'BARRIOS_HECHO_EL CINAL', 'BARRIOS_HECHO_EL CRISTAL', 'BARRIOS_HECHO_EL DIVISO', 'BARRIOS_HECHO_EL DORADO', 'BARRIOS_HECHO_EL FONCE', 'BARRIOS_HECHO_EL GALLINERAL', 'BARRIOS_HECHO_EL JARDIN', 'BARRIOS_HECHO_EL LLANITO', 'BARRIOS_HECHO_EL PABLON', 'BARRIOS_HECHO_EL PABLON ( LA TORRE )', 'BARRIOS_HECHO_EL PABLON ( VILLA LINA )', 'BARRIOS_HECHO_EL PABLON ( VILLA PATRICIA )', 'BARRIOS_HECHO_EL PALENQUE', 'BARRIOS_HECHO_EL PARAISO', 'BARRIOS_HECHO_EL PORVENIR', 'BARRIOS_HECHO_EL PRADO', 'BARRIOS_HECHO_EL PROGRESO', 'BARRIOS_HECHO_EL REPOSO', 'BARRIOS_HECHO_EL RETIRO', 'BARRIOS_HECHO_EL ROCIO', 'BARRIOS_HECHO_EL ROSAL', 'BARRIOS_HECHO_EL TEJAR', 'BARRIOS_HECHO_EL UVO', 'BARRIOS_HECHO_ESPAÑA', 'BARRIOS_HECHO_ESPERANZA I', 'BARRIOS_HECHO_ESPERANZA II', 'BARRIOS_HECHO_ESPERANZA III', 'BARRIOS_HECHO_ESTORAQUES', 'BARRIOS_HECHO_FACULTAD DE MINAS U. NACIONAL C-7', 'BARRIOS_HECHO_FATIMA', 'BARRIOS_HECHO_FONTANA', 'BARRIOS_HECHO_GAITAN', 'BARRIOS_HECHO_GALAN', 'BARRIOS_HECHO_GALLINERAL', 'BARRIOS_HECHO_GARCIA ROVIRA', 'BARRIOS_HECHO_GIRARDOT', 'BARRIOS_HECHO_GOMEZ NIÑO', 'BARRIOS_HECHO_GRANADA', 'BARRIOS_HECHO_GRANJAS DE JULIO RINCON', 'BARRIOS_HECHO_GRANJAS DE PROVENZA', 'BARRIOS_HECHO_GRANJAS REGAN', 'BARRIOS_HECHO_GRANJAS RIGAN', 'BARRIOS_HECHO_GUADUALES', 'BARRIOS_HECHO_GUAYACANES', 'BARRIOS_HECHO_HACIENDA DEL CACIQUE', 'BARRIOS_HECHO_HACIENDA SAN JUAN', 'BARRIOS_HECHO_HOYO CHIQUITO', 'BARRIOS_HECHO_IGSABELAR', 'BARRIOS_HECHO_INEM', 'BARRIOS_HECHO_INV. CAMINOS DE PAZ', 'BARRIOS_HECHO_INV. CARLOS PIZARRO', 'BARRIOS_HECHO_INV. CLUB CHIMITA', 'BARRIOS_HECHO_INV. DELICIAS', 'BARRIOS_HECHO_INV. LOS CORRALES', 'BARRIOS_HECHO_INV. LUZ DE ESPERANZA', 'BARRIOS_HECHO_INV. LUZ DE SALVACION', 'BARRIOS_HECHO_INV. NUEVO HORIZONTE', 'BARRIOS_HECHO_INV. PAISAJES DEL NORTE', 'BARRIOS_HECHO_INV. PUENTE NARIÑO', 'BARRIOS_HECHO_INV. PUNTA BETIN', 'BARRIOS_HECHO_INV. QUEBRADA LA IGLESIA', 'BARRIOS_HECHO_INV. SERVUNION', 'BARRIOS_HECHO_INV. TIBURONES', 'BARRIOS_HECHO_INV. VILLAS DEL GIRARDOT', 'BARRIOS_HECHO_INVASION GALLINERAL', 'BARRIOS_HECHO_JARDINES DE ALTA GRACIA', 'BARRIOS_HECHO_JARDINES DE COAVICONSA', 'BARRIOS_HECHO_JOSE ANTONIO GALAN', 'BARRIOS_HECHO_JOSE MARIA CORDOBA', 'BARRIOS_HECHO_JUAN XXIII', 'BARRIOS_HECHO_KENEDDY', 'BARRIOS_HECHO_KENNEDY', 'BARRIOS_HECHO_LA AURORA', 'BARRIOS_HECHO_LA CAMPIÑA', 'BARRIOS_HECHO_LA CEIBA', 'BARRIOS_HECHO_LA CEMENTO', 'BARRIOS_HECHO_LA CONCORDIA', 'BARRIOS_HECHO_LA ESTRELLA', 'BARRIOS_HECHO_LA FERIA', 'BARRIOS_HECHO_LA FLORA', 'BARRIOS_HECHO_LA FLORESTA', 'BARRIOS_HECHO_LA FORTUNA', 'BARRIOS_HECHO_LA GLORIA', 'BARRIOS_HECHO_LA GRAN LADERA', 'BARRIOS_HECHO_LA GRANJITA', 'BARRIOS_HECHO_LA INDEPENDENCIA', 'BARRIOS_HECHO_LA INMACULADA', 'BARRIOS_HECHO_LA ISLA', 'BARRIOS_HECHO_LA JOYA', 'BARRIOS_HECHO_LA JUVENTUD', 'BARRIOS_HECHO_LA LIBERTAD', 'BARRIOS_HECHO_LA MALAÑA (URBANO)', 'BARRIOS_HECHO_LA PALMA', 'BARRIOS_HECHO_LA PEDREGOSA', 'BARRIOS_HECHO_LA QUEBRADA', 'BARRIOS_HECHO_LA ROSITA', 'BARRIOS_HECHO_LA SALLE', 'BARRIOS_HECHO_LA UNIVERSIDAD', 'BARRIOS_HECHO_LA VICTORIA', 'BARRIOS_HECHO_LAGOS DEL CACIQUE', 'BARRIOS_HECHO_LAS AMERICAS', 'BARRIOS_HECHO_LAS CASITAS', 'BARRIOS_HECHO_LAS COLINAS', 'BARRIOS_HECHO_LAS DELICIAS', 'BARRIOS_HECHO_LAS GRANJITAS B/GA', 'BARRIOS_HECHO_LAS GUACAMAYAS', 'BARRIOS_HECHO_LAS HAMACAS', 'BARRIOS_HECHO_LAS MARGARITAS MUTIS', 'BARRIOS_HECHO_LAURELES', 'BARRIOS_HECHO_LIMONCITO', 'BARRIOS_HECHO_LIZCANO I', 'BARRIOS_HECHO_LIZCANO II', 'BARRIOS_HECHO_LOS ACASIOS', 'BARRIOS_HECHO_LOS ANGELES', 'BARRIOS_HECHO_LOS CAMBULOS', 'BARRIOS_HECHO_LOS CANELOS', 'BARRIOS_HECHO_LOS CEDROS', 'BARRIOS_HECHO_LOS CENTAUROS', 'BARRIOS_HECHO_LOS COMUNEROS', 'BARRIOS_HECHO_LOS GUADUALES', 'BARRIOS_HECHO_LOS PINOS', 'BARRIOS_HECHO_LOS PORTALES II', 'BARRIOS_HECHO_LOS ROBLES', 'BARRIOS_HECHO_LOS ROSALES', 'BARRIOS_HECHO_LOS SAUCES', 'BARRIOS_HECHO_MALPASO', 'BARRIOS_HECHO_MANUELA BELTRAN', 'BARRIOS_HECHO_MANUELA BELTRAN I', 'BARRIOS_HECHO_MANUELA BELTRAN II', 'BARRIOS_HECHO_MANZANARES', 'BARRIOS_HECHO_MARIA AUXILIADORA', 'BARRIOS_HECHO_MARIA PAZ', 'BARRIOS_HECHO_MARIANELA', 'BARRIOS_HECHO_MEJORAS PUBLICAS', 'BARRIOS_HECHO_MERCEDES', 'BARRIOS_HECHO_MESETAS DEL SANTUARIO', 'BARRIOS_HECHO_METROPOLITANA', 'BARRIOS_HECHO_MILAGRO DE DIOS', 'BARRIOS_HECHO_MINUTO DE DIOS', 'BARRIOS_HECHO_MIRADOR', 'BARRIOS_HECHO_MIRADOR DE COLORADOS', 'BARRIOS_HECHO_MIRADOR DE SAN JUAN', 'BARRIOS_HECHO_MIRADOR DEL CACIQUE', 'BARRIOS_HECHO_MIRADOR DEL NORTE', 'BARRIOS_HECHO_MIRADORES DE SAN LORENZO', 'BARRIOS_HECHO_MIRADORES DEL KENNEDY', 'BARRIOS_HECHO_MIRAFLORES', 'BARRIOS_HECHO_MIRAMAR', 'BARRIOS_HECHO_MODELO', 'BARRIOS_HECHO_MONEQUES', 'BARRIOS_HECHO_MONEQUEZ', 'BARRIOS_HECHO_MORAVIA  C-4', 'BARRIOS_HECHO_MORRORICO', 'BARRIOS_HECHO_MUTIS', 'BARRIOS_HECHO_MUTUALIDAD', 'BARRIOS_HECHO_NAPOLES', 'BARRIOS_HECHO_NARIÑO', 'BARRIOS_HECHO_NAVAS', 'BARRIOS_HECHO_NIZA', 'BARRIOS_HECHO_NORTE BAJO', 'BARRIOS_HECHO_NUEVA COLOMBIA', 'BARRIOS_HECHO_NUEVA GRANADA', 'BARRIOS_HECHO_NUEVO COLON', 'BARRIOS_HECHO_NUEVO SOTOMAYOR', 'BARRIOS_HECHO_OLAS ALTAS', 'BARRIOS_HECHO_OLAS BAJAS', 'BARRIOS_HECHO_OLAS II', 'BARRIOS_HECHO_OMAGA (SECTOR I )', 'BARRIOS_HECHO_OMAGA (SECTOR II )', 'BARRIOS_HECHO_OMAGA (SECTOR III )', 'BARRIOS_HECHO_PABLO VI', 'BARRIOS_HECHO_PABLON ALTO', 'BARRIOS_HECHO_PABLON BAJO', 'BARRIOS_HECHO_PALMERAS DEL CACIQUE', 'BARRIOS_HECHO_PAN DE AZUCAR', 'BARRIOS_HECHO_PANTANO I', 'BARRIOS_HECHO_PANTANO II', 'BARRIOS_HECHO_PANTANO III', 'BARRIOS_HECHO_PARQUE INDUSTRIAL', 'BARRIOS_HECHO_PATIO BONITO (COLORADOS)', 'BARRIOS_HECHO_PAYADOR REAL', 'BARRIOS_HECHO_PEÑON DEL VALLE', 'BARRIOS_HECHO_PIO XII', 'BARRIOS_HECHO_POBLADO', 'BARRIOS_HECHO_PORTAL DEL CACIQUE', 'BARRIOS_HECHO_PORTON DEL TEJAR', 'BARRIOS_HECHO_PORVENIR', 'BARRIOS_HECHO_PRIMAVERAL', 'BARRIOS_HECHO_PRIMERO DE MAYO', 'BARRIOS_HECHO_PROVENZA', 'BARRIOS_HECHO_PUERTA DEL SOL', 'BARRIOS_HECHO_PUERTO MADERO', 'BARRIOS_HECHO_PUERTO RICO', 'BARRIOS_HECHO_PUNTA ESTRELLA', 'BARRIOS_HECHO_PUNTA PARAISO', 'BARRIOS_HECHO_QUEBRADA LA IGLESIA', 'BARRIOS_HECHO_QUINTA DANIA', 'BARRIOS_HECHO_QUINTA ESTRELLA', 'BARRIOS_HECHO_QUINTA GRANADA', 'BARRIOS_HECHO_QUINTADANIA', 'BARRIOS_HECHO_QUINTAS DEL CACIQUE', 'BARRIOS_HECHO_REGADERO NORTE', 'BARRIOS_HECHO_RICAURTE', 'BARRIOS_HECHO_RINCON DE GIRON', 'BARRIOS_HECHO_RINCON DE LA PAZ', 'BARRIOS_HECHO_RIO DE ORO II', 'BARRIOS_HECHO_RUITOQUE BAJO', 'BARRIOS_HECHO_SAN ALONSO', 'BARRIOS_HECHO_SAN CRISTOBAL', 'BARRIOS_HECHO_SAN FRANCISCO', 'BARRIOS_HECHO_SAN FRANCISCO P/TA', 'BARRIOS_HECHO_SAN GERARDO', 'BARRIOS_HECHO_SAN IGNACIO', 'BARRIOS_HECHO_SAN LUIS', 'BARRIOS_HECHO_SAN MARTIN', 'BARRIOS_HECHO_SAN MIGUEL', 'BARRIOS_HECHO_SAN PEDRITO', 'BARRIOS_HECHO_SAN PEDRO CLAVER', 'BARRIOS_HECHO_SAN RAFAEL', 'BARRIOS_HECHO_SAN RAFAEL P/TA', 'BARRIOS_HECHO_SAN VALENTIN', 'BARRIOS_HECHO_SANTA MARIA', 'BARRIOS_HECHO_SANTA MONICA DEL TEJAR', 'BARRIOS_HECHO_SANTANDER', 'BARRIOS_HECHO_SERREZUELA', 'BARRIOS_HECHO_SIN INFORMACION', 'BARRIOS_HECHO_SOTOMAYOR', 'BARRIOS_HECHO_TEJADOS', 'BARRIOS_HECHO_TEJAR NORTE', 'BARRIOS_HECHO_TEJAR NORTE (SECTOR I )', 'BARRIOS_HECHO_TEJAR NORTE (SECTOR II )', 'BARRIOS_HECHO_TERMINAL DE TRANSPORTES', 'BARRIOS_HECHO_TERRAZAS', 'BARRIOS_HECHO_TESOROS DEL CACIQUE', 'BARRIOS_HECHO_TOLEDO PLATA', 'BARRIOS_HECHO_TRANSICION', 'BARRIOS_HECHO_TRANSICION I', 'BARRIOS_HECHO_TRANSICION II', 'BARRIOS_HECHO_TRANSICION III', 'BARRIOS_HECHO_TRANSICION IV', 'BARRIOS_HECHO_TRANSICION V', 'BARRIOS_HECHO_UNIVERSIDAD', 'BARRIOS_HECHO_URB VILLAS DE SAN PATRICIO', 'BARRIOS_HECHO_URB. ACROPOLIS I', 'BARRIOS_HECHO_URB. ACROPOLIS II', 'BARRIOS_HECHO_URB. ACROPOLIS III', 'BARRIOS_HECHO_URB. ALTOS DE TAJAMAR', 'BARRIOS_HECHO_URB. BALCONCITOS', 'BARRIOS_HECHO_URB. BALCONES DE LA COLINA', 'BARRIOS_HECHO_URB. BALCONES DE PROVENZA', 'BARRIOS_HECHO_URB. BAVARIA', 'BARRIOS_HECHO_URB. BRISAS DE PROVENZA', 'BARRIOS_HECHO_URB. BRISAS DEL MUTIS', 'BARRIOS_HECHO_URB. CIUDAD BOLIVAR', 'BARRIOS_HECHO_URB. COLSEGUROS NORTE', 'BARRIOS_HECHO_URB. EL SOL I', 'BARRIOS_HECHO_URB. EL SOL II', 'BARRIOS_HECHO_URB. ESTORAQUES I', 'BARRIOS_HECHO_URB. ESTORAQUES II', 'BARRIOS_HECHO_URB. FATIMA', 'BARRIOS_HECHO_URB. FUNDADORES', 'BARRIOS_HECHO_URB. GETSEMANI', 'BARRIOS_HECHO_URB. HEROES', 'BARRIOS_HECHO_URB. LA GRAN LADERA', 'BARRIOS_HECHO_URB. LA HOYADA', 'BARRIOS_HECHO_URB. LOS ALMENDROS', 'BARRIOS_HECHO_URB. LOS NARANJOS', 'BARRIOS_HECHO_URB. LOS SAUCES', 'BARRIOS_HECHO_URB. MACAREGUA', 'BARRIOS_HECHO_URB. MANZANARES', 'BARRIOS_HECHO_URB. MARSELLA REAL', 'BARRIOS_HECHO_URB. MIRADOR DE FATIMA', 'BARRIOS_HECHO_URB. MIRADORES DE LA UIS', 'BARRIOS_HECHO_URB. MONTE REDONDO', 'BARRIOS_HECHO_URB. MONTE VERDE', 'BARRIOS_HECHO_URB. NEPTUNO', 'BARRIOS_HECHO_URB. PASEO LA FERIA', 'BARRIOS_HECHO_URB. PLAZA MAYOR', 'BARRIOS_HECHO_URB. PLAZUELA REAL', 'BARRIOS_HECHO_URB. PORTOFINO', 'BARRIOS_HECHO_URB. PRADOS DEL MUTIS', 'BARRIOS_HECHO_URB. QUINTAS DE SANTANA', 'BARRIOS_HECHO_URB. RINCON DEL LAGO', 'BARRIOS_HECHO_URB. ROSA ALTA', 'BARRIOS_HECHO_URB. SAMANES', 'BARRIOS_HECHO_URB. SAN FERMIN', 'BARRIOS_HECHO_URB. SAN VALENTIN', 'BARRIOS_HECHO_URB. TERRANOVA', 'BARRIOS_HECHO_URB. TORRES DE ALEJANDRÍA', 'BARRIOS_HECHO_URB. TORRES DE MONTERREY', 'BARRIOS_HECHO_URB. VIVEROS DE PROVENZA', 'BARRIOS_HECHO_URBANIZACION QUINTAS DEL CACIQUE', 'BARRIOS_HECHO_URBANIZACION SANTA BARBARA', 'BARRIOS_HECHO_VDA. 10 DE MAYO', 'BARRIOS_HECHO_VDA. AGUABLANCA', 'BARRIOS_HECHO_VDA. ALTO DE LOS PADRES', 'BARRIOS_HECHO_VDA. ANGELINOS', 'BARRIOS_HECHO_VDA. ARGELIA', 'BARRIOS_HECHO_VDA. BOLARQUI', 'BARRIOS_HECHO_VDA. CUCHILLA ALTA', 'BARRIOS_HECHO_VDA. EL ABURRIDO', 'BARRIOS_HECHO_VDA. EL NOGAL (COLORADOS)', 'BARRIOS_HECHO_VDA. GUALILO ALTO', 'BARRIOS_HECHO_VDA. GUALILO BAJO', 'BARRIOS_HECHO_VDA. LA CAPILLA', 'BARRIOS_HECHO_VDA. LA ESMERALDA', 'BARRIOS_HECHO_VDA. LA ESPERANZA', 'BARRIOS_HECHO_VDA. LA MALAÑA', 'BARRIOS_HECHO_VDA. LA SABANA', 'BARRIOS_HECHO_VDA. LAS MARIAS', 'BARRIOS_HECHO_VDA. LOS CUADROS', 'BARRIOS_HECHO_VDA. LOS SANTOS', 'BARRIOS_HECHO_VDA. MAGUEYES', 'BARRIOS_HECHO_VDA. MALPASO', 'BARRIOS_HECHO_VDA. MONSERRATE', 'BARRIOS_HECHO_VDA. PEDREGAL', 'BARRIOS_HECHO_VDA. RETIRO CHIQUITO', 'BARRIOS_HECHO_VDA. RETIRO GRANDE', 'BARRIOS_HECHO_VDA. RIO DE ORO', 'BARRIOS_HECHO_VDA. ROSA BLANCA', 'BARRIOS_HECHO_VDA. SAN CAYETANO', 'BARRIOS_HECHO_VDA. SAN IGNACIO', 'BARRIOS_HECHO_VDA. SAN JOSE', 'BARRIOS_HECHO_VDA. SAN PEDRO ALTO', 'BARRIOS_HECHO_VDA. SAN PEDRO BAJO', 'BARRIOS_HECHO_VDA. SANTA BARBARA', 'BARRIOS_HECHO_VDA. SANTA RITA', 'BARRIOS_HECHO_VDA. TRES ESQUINAS', 'BARRIOS_HECHO_VDA. VERICUTE', 'BARRIOS_HECHO_VDA. VIJAGUAL', 'BARRIOS_HECHO_VEGAS DE MORRORICO', 'BARRIOS_HECHO_VEGAS DE MORRORRICO', 'BARRIOS_HECHO_VEREDA EL PABLON', 'BARRIOS_HECHO_VEREDA MARCELITAS', 'BARRIOS_HECHO_VILLA ALEGRIA', 'BARRIOS_HECHO_VILLA ALICIA', 'BARRIOS_HECHO_VILLA CANDADO', 'BARRIOS_HECHO_VILLA DE LOS CABALLEROS', 'BARRIOS_HECHO_VILLA DE LOS CONQUISTADORES', 'BARRIOS_HECHO_VILLA DEL NOGAL', 'BARRIOS_HECHO_VILLA DEL PRADO', 'BARRIOS_HECHO_VILLA ESMERALDA', 'BARRIOS_HECHO_VILLA FLOR', 'BARRIOS_HECHO_VILLA HELENA', 'BARRIOS_HECHO_VILLA HELENA I', 'BARRIOS_HECHO_VILLA HELENA II', 'BARRIOS_HECHO_VILLA INES', 'BARRIOS_HECHO_VILLA MARIA', 'BARRIOS_HECHO_VILLA MARIA III (BAJA)', 'BARRIOS_HECHO_VILLA MERCEDES', 'BARRIOS_HECHO_VILLA REAL', 'BARRIOS_HECHO_VILLA ROMERO', 'BARRIOS_HECHO_VILLA ROSA', 'BARRIOS_HECHO_VILLA ROSA (SECTOR II )', 'BARRIOS_HECHO_VILLA ROSA (SECTOR III )', 'BARRIOS_HECHO_VILLA SARA', 'BARRIOS_HECHO_VILLABEL', 'BARRIOS_HECHO_VILLAS DE SAN FRANCISCO', 'BARRIOS_HECHO_VILLAS DE SAN IGNACIO', 'BARRIOS_HECHO_VILLAS DEL NOGAL', 'BARRIOS_HECHO_VRDA  ABEJAS', 'BARRIOS_HECHO_VRDA  CAPILLA', 'BARRIOS_HECHO_VRDA  JABONCILLO', 'BARRIOS_HECHO_VRDA  SAN ISIDRO', 'BARRIOS_HECHO_ZARABANDA', 'BARRIOS_HECHO_ZONA INDUSTRIAL')] not found in axis"